src/Handlers/AbstractPaymentMethodHandler.php

```diff
@@ -686,4 +682,32 @@ private function getProduct(string $productId, Context $context): ProductEntity
 
         return $product;
     }
+
+    /**
+     * @param PaymentException $exception
+     * @param $transactionId
+     */
+    private function handlePaymentException(PaymentException $exception, $transactionId)
+    {
+
+        $exceptionType = get_class($exception);
+
+        switch ($exceptionType) {
+            case PaymentCancelledException::class:
+                throw new CustomerCanceledAsyncPaymentException(
+                    $transactionId,
+                    $exception->getMessage()
+                );
+            case PaymentFailedException::class:
+                throw new AsyncPaymentFinalizeException(
+                    $transactionId,
+                    $exception->getMessage()
+                );
+            default:
+                throw new AsyncPaymentFinalizeException(
+                    $transactionId,
+                    $exception->getMessage()
+                );
```

Comment:
Should we add a break; and combine the second alternative with default? Needs testing though 
```suggestion
                break;
            case PaymentFailedException::class:
            default:
                throw new AsyncPaymentFinalizeException(
                    $transactionId,
                    $exception->getMessage()
                );
```

Reply:
Looks good to me